cpu/esp32/esp-ieee802154/esp_ieee802154_hal.c

```diff
@@ -0,0 +1,466 @@
+/*
+ * Copyright (C) 2025 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include "esp_ieee802154.h"
+
+#include "esp_ieee802154_hal.h"
+#include "log.h"
+#include "net/ieee802154/radio.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+/* ACK frame timeout in microseconds, should be a multiple of 16 */
+#define ESP_IEEE802154_ACK_TIMEOUT_US (3456)
+
+_Static_assert((3456 % 16 == 0), "ACK frame timeout should be a multiple of 16");
+
+/* Although the device driver supports IEEE802154_CAP_IRQ_TX_START,
+ * IEEE802154_CAP_IRQ_RX_START and IEEE802154_CAP_IRQ_CCA_DONE, we are not
+ * using it for now to avoid unnecessary performance degradation as it is
+ * not used by any link layer driver */
+#define _USE_RX_START   0
+#define _USE_TX_START   0
+#define _USE_CCA_DONE   0
+
+#define _USE_SET_RX     1
+#define _USE_SET_IDLE   1
```

Comment:
Is there a case where those would not be set to 1?

Reply:
Good question, I used them to be able to enable/disable the corresponding code quickly. I would prefer to leave them there for future testing.